File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
        => speed switch
        {
            0 => 0,
            int s when s >= 1 && s <= 4 => 1,
            int s when s >= 5 && s <= 8 => .9,
            9 => .8,
            10 => .77,
            _ => throw new ArgumentException("Invalid speed value", nameof(speed)),
        };
    public static double ProductionRatePerHour(int speed) => speed * SuccessRate(speed) * 221.0;
    public static int WorkingItemsPerMinute(int speed) => (int)ProductionRatePerHour(speed) / 60;
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | 0 -> 0.0
    | s when s <= 4 -> 1.0
    | s when s <= 8 -> 0.9
    | s when s = 9 -> 0.8
    | s when s = 10 -> 0.77
    | _ -> 0.0

let productionRatePerHour (speed: int) : float =
    successRate (speed) * float (speed) * 221.0

let workingItemsPerMinute (speed: int) : int =
    int (productionRatePerHour (speed) / 60.0)
